dev-docs/007-transforms-rfc.md

```diff
@@ -0,0 +1,99 @@
+- Start Date: 2019-09-05
+- RFC PR: 
+
+# Summary
+
+It is desirable to enable arbitrary transform support common in robotic systems. While XVIZ models some explicitly
+with our coordinate types IDENTITY, GEOGRAPHIC, and VEHICLE_RELATIVE, the need for client controlled frames of reference
+will make it easier to interop with other systems as well as match a familiar modeling automous system data.
+
+# Motivation
+
+Some PERSISTENT data, such as *maps* and *point clouds*, could be encoded in various ways such as vehicle relative.
+If they are semantically fixed, but the reference point changes, there is no way to efficiently update them without
+sending all the points again.
+
+Ideally, the only data that is required is the transform to adjust how the points are transformed.
+
+Extending this to a general robotic systems convention, the need to describe parent/child relationships
+is a common model for robotic systems.
+
+XVIZ is not focused on articulated robotic platforms but support for the convention common when modeling autonomous data
+would be benefitial for XVIZ and it's users.
+
+# Proposal
+
+### POSE schema
+ - Add a *parent* field to the pose that is a string that my reference another pose stream
+   If present within the data, it would be used to construct the final transform matrix for the data
+ - Poses may reference any other pose as a *parent* across all state_update message types. Meaning a PERSISTENT
+   pose could be a parent of a *COMPLETE* or *INCREMENTAL* pose.
```

Comment:
*INCREMENTAL* pose. - if I understand it correctly, (i.e. sending a Pose that is a transform with respect to the previous pose, as opposed to an "arbitrary" starting pose) I see the appeal of this, but beware of numerical instability from repeated matrix multiplications. For longer logs the matrices may no longer be well-conditioned after some time. This would be very noticeable to users but could be almost impossible to solve short of a redesign. 

This is only an issue if having to compose many matrices (i.e. reconciling geometry in frames that are "far apart").

If the static streams will only exist for a little while (perhaps submaps that are loaded and shown for some time and then removed after the vehicle advances) it may be less severe.

I think it may be worth weaving some of this into the RFC text...

Reply:
I agree, I don't think *INCREMENTAL* poses would be of much value and I should call out specifically that is not the intention here.

The semantics of *INCREMENTAL* in XVIZ is to _add data to the current state_, rather than compose with the prior state.

Good clarification.